Sources/MapboxCoreNavigation/RouteController.swift

```diff
@@ -323,19 +342,14 @@ open class RouteController: NSObject {
         
         updateIndexes(status: status, progress: routeProgress)
         updateRouteLegProgress(status: status)
-        let willReroute = !userIsOnRoute(snappedLocation, status: status)
-        && (delegate?.router(self, shouldRerouteFrom: snappedLocation)
-            ?? DefaultBehavior.shouldRerouteFromLocation)
         
-        updateSpokenInstructionProgress(status: status, willReRoute: willReroute)
+        updateSpokenInstructionProgress(status: status, willReRoute: isRerouting)
         updateVisualInstructionProgress(status: status)
         updateRoadName(status: status)
         updateDistanceToIntersection(from: snappedLocation)
         
-        if willReroute {
-            reroute(from: snappedLocation, along: routeProgress)
-        }
-
+        rerouteAfterArrivalIfNeeded(snappedLocation, status: status)
```

Comment:
Have you got a chance to test if it still works with all the changes?

Reply:
I didn't do test drives, but tests passed...

Reply:
After closer testing I found that indeed it doesn't work, but our mechanism works correctly. The problem is that NN's `forceReroute` does not actually trigger a reroute as expected. And it is not related to arriving or if it was called mid-route. Will need a fix on that from @mapbox/navnative

Reply:
After calling `RerouteControllerInterface.forceReroute()`, `RerouteObserver` does not receive any messages about detected, received or failed reroutes (while reroutes, triggered by user wandering away from the route line, works correct). At first I thought that it ignores the reroute call when `state` is completed, but forcing a reroute in the middle of the drive also did not result in any `RerouteObserver` calls. I checked it on NN v94.0.0

Reply:
@Udumft just to clarify: `onRerouteDetected` is not called too?

Reply:
`onRerouteDetected` was not called too.

UPD.
This issue should be resolved in next NN release

Reply:
It is. Latest NN release has the required fix and I verified it works correctly.